models/Community.js

```diff
@@ -3,16 +3,47 @@ module.exports = (sequelize, DataTypes) => {
         name: {
             type: DataTypes.STRING,
             allowNull: false,
+            default: '',
             validate: {
-                isAlphanumeric: true,
-                len: [1, 25]
+                notEmpty: {
+                    msg: 'The community name must be filled in'
+                },
+                isAlphanumeric: {
+                    msg: 'The community name must be alphanumeric without spaces'
+                },
+                len: {
+                    args: [1, 25],
+                    msg: 'The community name must be between 1 and 25 characters'
+                },
+                isUnique: function(name, done) {
+                    // Used to improve the error message as a pre validation step instead of an index error
+                    Community.findOne({
+                        where: sequelize.where(sequelize.fn('lower', sequelize.col('name')), name.toLowerCase()),
+                    }).then((name) => {
+                        if (!name) {
+                            done();
+                        }
+                        else {
+                            done(new Error('That community name is already in use'));
+                        }
+                    }).catch((err) => {
+                        done(err);
+                    });
+                }
             }
         },
         title: {
             type: DataTypes.STRING,
             allowNull: false,
+            default: '',
             validate: {
-                len: [1, 100]
+                notEmpty: {
+                    msg: 'The community title must be filled in'
+                },
+                len: {
+                    args: [1, 100],
+                    msg: 'The title must be between 1 and 100 characters'
```

Comment:
These are also enforced in the community creation page.

Reply:
Answered above.